### PR TITLE
Added casts properties to all models, re-ordered model properties

### LINKED ISSUE #916: Status changes when editing component
If the status of a component is not 'Operational' and you edit it then save, it will be set to operational unless you remember to change the loaded 'default'.

The 'Status' field in the component edit should reflect the current status, not simply always set to 'Operational' as it is nasssty if you change a detail field for a component and its status changes when you didn't mean it to do so.

### PR DESCRIPTION
**Pull request by jbrooksuk**
Should fix #916

### CODE REVIEW

File: app/Models/Component.php [hunk 1]
```diff
@@ -23,17 +23,37 @@ class Component extends Model implements HasPresenter
...
+     * The attributes that should be casted to native types.
      *
      * @var string[]
      */
-    public $rules = [
-        'name'   => 'required|string',
-        'status' => 'integer|required',
-        'link'   => 'url',
+    protected $casts = [
+        'order'       => 'integer',
```
Comment (GrahamCampbell) [inline app/Models/Component.php]: int

File: app/Models/Component.php [hunk 2]
```diff
@@ -23,17 +23,37 @@ class Component extends Model implements HasPresenter
...
+        'group_id'    => 'integer',
```
Comment (GrahamCampbell) [inline app/Models/Component.php]: int

File: app/Models/Incident.php [hunk 3]
```diff
@@ -23,19 +23,30 @@ class Incident extends Model implements HasPresenter
...
-        'message'      => 'required',
+    protected $appends = ['human_status'];
+
+    /**
+     * The attributes that should be casted to native types.
+     *
+     * @var string[]
+     */
+    protected $casts = [
+        'visible'      => 'integer',
```
Comment (GrahamCampbell) [inline app/Models/Incident.php]: int

File: app/Models/Metric.php [hunk 4]
```diff
@@ -37,16 +37,16 @@ class Metric extends Model implements HasPresenter
...
      */
-    public $rules = [
-        'name'          => 'required',
-        'suffix'        => 'required',
+    protected $casts = [
+        'name'          => 'string',
         'display_chart' => 'boolean',
-        'default_value' => 'numeric',
-        'places'        => 'numeric|min:0|max:4',
+        'default_value' => 'integer',
```
Comment (GrahamCampbell) [inline app/Models/Metric.php]: in all places, please use the correct types

File: app/Models/User.php [hunk 5]
```diff
@@ -25,17 +25,27 @@ class User extends Model implements AuthenticatableContract, CanResetPasswordCon
...
-    public $rules = [
-        'username' => ['required', 'regex:/\A(?!.*[:;]-\))[ -~]+\z/'],
-        'email'    => 'required|email',
-        'password' => 'required',
+    protected $casts = [
+        'username'          => 'string',
+        'email'             => 'string',
+        'google_2fa_secret' => 'string',
+        'api_key'           => 'string',
+        'active'            => 'boolean',
```
Comment (GrahamCampbell) [inline app/Models/User.php]: bool

File: app/Models/Metric.php (hunk 4 above)
Comment (jbrooksuk) [inline app/Models/Metric.php]: I didn't think Laravel supported short versions?

Comment (GrahamCampbell) [inline app/Models/Metric.php]: it does

Comment (jbrooksuk) [inline app/Models/Metric.php]: Since when? :)

**Comment by GrahamCampbell** [comment]
Shouldn't this go to 1.2 since it's a bug fix?

**Comment by jbrooksuk** [comment]
1.2.1 I guess...

**Comment by jbrooksuk** [comment]
Rebased and tests do pass.

**Comment by GrahamCampbell** [comment]
> Rebased and tests do pass.

Still failing?

**Comment by jbrooksuk** [comment]
Hmm, leave it with me.

**Comment by jbrooksuk** [comment]
This actually fixes the tests too, since they weren't comparing integers.

**Comment by joecohens** [comment]
:+1:

**Pull request merged**